File: 0153-寻找旋转排序数组中的最小值/Solution.java
class Solution {
    public int findMin(int[] nums) {
        if(nums.length == 1){ return nums[0];}
        int left=0, right=nums.length-1;
        while(left < right){
            int mid = (left+right)/2;
            if(nums[mid] < nums[right]){
                right = mid;
            }else{
                left = mid+1;
            }
        }
        return nums[left];
    }
}

File: 0153-寻找旋转排序数组中的最小值/solution.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        i, j = 0, len(nums)-1
        while i < j:
            mid = (i + j) // 2
            if nums[mid] > nums[j]:
                i = mid + 1
            elif nums[mid] < nums[j]:
                j = mid
        return nums[i]